src/store/root-state.epics.ts

```diff
@@ -1,15 +1,26 @@
 import Keychain from 'react-native-keychain';
 import { combineEpics } from 'redux-observable';
-import { EMPTY, forkJoin, from, Observable } from 'rxjs';
-import { concatMap, mapTo, switchMap, withLatestFrom } from 'rxjs/operators';
+import { EMPTY, forkJoin, from, Observable, of } from 'rxjs';
+import { concatMap, filter, mapTo, switchMap, withLatestFrom } from 'rxjs/operators';
 import { Action } from 'ts-action';
 import { ofType, toPayload } from 'ts-action-operators';
 
 import { BeaconHandler } from '../beacon/beacon-handler';
 import { globalNavigationRef } from '../navigator/root-stack';
+import { isDefined } from '../utils/is-defined';
 import { getKeychainOptions } from '../utils/keychain.utils';
 import { RootState } from './create-store';
-import { rootStateResetAction, untypedNavigateAction } from './root-state.actions';
+import { isFirstAppLaunchCheckAction, rootStateResetAction, untypedNavigateAction } from './root-state.actions';
+import { setIsReinstalled } from './settings/settings-actions';
+
+const reinstallEpic = (action$: Observable<Action>, state$: Observable<RootState>) =>
+  action$.pipe(
+    ofType(isFirstAppLaunchCheckAction.submit),
+    withLatestFrom(state$, (_, state) => state.settings.isNotFirstAppLaunch),
+    switchMap(isNotFirstAppLaunch => of(isDefined(isNotFirstAppLaunch) || isNotFirstAppLaunch === true)),
```

Comment:
you could change  `switchMap` to `map` and remove `of()` (you will be able to return synchronous result)

Reply:
you even could just omit `map` here
```
filter(isFirstAppLaunch => isFirstAppLaunch === true)
```